### ISSUE TITLE
Can't load GRUCell weights from TensorFlow 1 to TensorFlow 2

### ISSUE THREAD

**Issue by paultbd** [tensorflow/tensorflow#44948]
# Issue 💻 

## System information
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): `True`
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `Windows 10`
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: `False`
- TensorFlow installed from (source or binary): `pip install tensorflow`
- TensorFlow version (use command below): `v2.3.0-rc2-23-gb36436b087 2.3.0`
- Python version: `3.8.3`
- Bazel version (if compiling from source): `None`
- GCC/Compiler version (if compiling from source): `None`
- CUDA/cuDNN version: `None`
- GPU model and memory: `None`

## Current Behavior

I migrated a code written with **tensorflow 1** to **tensorflow 2**. In addition I would like to convert the checkpoints of my **tensorflow 1** model too. However, the `GRUCell` seems really different from **tensorflow 1** to **tensorflow 2** implementation, and can't load the saved weights from a `compat.v1.nn.rnn_cell.GRUCell` to a `keras.layers.GRUCell`.
The error is:
```
W tensorflow/core/framework/op_kernel.cc:1767] OP_REQUIRES failed at save_restore_v2_ops.cc:184 : Not found: Key gru_cell_1/bias:0 not found in checkpoint
```

So the error comes from the weights initialization. By looking closer to the structure of the `GRUCell` from the **1** and **2** versions, I noticed major differences:

1. The `compat.v1.rnn_cell.GRUCell` has **four weights**:
    * `gru_cell/gates/kernel:0` of shape `(S + H, 2 x H)`, 
    * `gru_cell/gates/bias:0` of shape `(2 x H, )`, 
    * `gru_cell/candidate/kernel:0` of shape `(S + H, H)`,  
    * `gru_cell/candidate/bias:0` of shape `(H, )`

2. The `tf.keras.layers.GRUCell` has **three weights**:
    * `gru_cell/kernel:0` of shape `(S, 3 x H)`
    * `gru_cell/recurrent_kernel:0` of shape `(H, 3 x H)`
    * `gru_cell/bias:0` of shape `(2, 3 x H)`

So the issue is that **tensorflow** struggles to load weights from the first version to a `keras.layers.GRUCell` - of course, the shape and length of the weights are different. 
Is there a function to migrate `compat.v1.nn.rnn_cell.GRUCell` weights to `keras.layers.GRUCell` ? I have not seen in the documentation.

### 1. Weights of `compat.v1.nn.rnn_cell.GRUCell` layer

```python
import tensorflow as tf

SEQ_LENGTH = 4
HIDDEN_SIZE = 512
BATCH_SIZE = 1
inputs = tf.random.normal([BATCH_SIZE, SEQ_LENGTH])

# GRU cell
gru = tf.compat.v1.nn.rnn_cell.GRUCell(HIDDEN_SIZE)
# Hidden state
state = gru.zero_state(BATCH_SIZE, tf.float32)
# Forward
output, state = gru(inputs, state)

for weight in gru.weights:
    print(weight.name, weight.shape)
```

Output:
```raw
gru_cell/gates/kernel:0 (516, 1024)
gru_cell/gates/bias:0 (1024,)
gru_cell/candidate/kernel:0 (516, 512)
gru_cell/candidate/bias:0 (512,)
```

### 2. Weights of `keras.layers.GRUCell` layer

```python
import tensorflow as tf

SEQ_LENGTH = 4
HIDDEN_SIZE = 512
BATCH_SIZE = 1
inputs = tf.random.normal([BATCH_SIZE , SEQ_LENGTH])

# GRU cell
gru = tf.keras.layers.GRUCell(HIDDEN_SIZE)
# Hidden state
state = tf.zeros((BATCH_SIZE, HIDDEN_SIZE), dtype=tf.float32)
# Forward
output, state = gru(inputs, state)

# Display the weigths
for weight in gru.weights:
    print(weight.name, weight.shape)
```

Output:
```raw
gru_cell/kernel:0 (4, 1536)
gru_cell/recurrent_kernel:0 (512, 1536)
gru_cell/bias:0 (2, 1536)
```

## Note

I tried [`_convert_rnn_weights` tensorflow function](https://github.com/tensorflow/tensorflow/blob/c52ee536c045186476450cf76461d47fb2667e7d/tensorflow/python/keras/saving/hdf5_format.py#L404-L568) to convert the desired weights. It works but only for `CuDNN` weights, so I can't use it in my case.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/f9aabf8ec47e727fbde26a297a4adaf0/44948.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/44cc623d82e2ad35605b21843094e035/44948-tf-nightly.ipynb). Please find the attached gist. Thanks!

**Comment by qlzh727** [comment]
Thanks for reporting the issue.

In short, the weights between compat.v1.nn.rnn_cell.GRUCell and keras.layers.GRUCell are not compatible between each other. We don't have a function to convert between them, and if you really want to do it, you will need to do it manually.

Math wise, if you have the numpy value of the v1 weights, the formula are:
B = batch_size
H = state_size

1. all_kernel = np.concat([gru_cell/gates/kernel, gru_cell/candidate/kernel], axis=1)  # shape (B+H, 3 * H)
2. kernel = all_kernel[:B]  # shape(B, 3 * H)
3. recurrent_kernel = all_kernel[B:] # shape (H, 3 * H)
4. bias = np.concat([gru_cell/gates/bias, gru_cell/candidate/bias], axis=0)  # shape (B, 3 * H)
5. zero_bias = np.zeros([B, 3 * H])
6. bias = np.concat([bias, zero_bias], axis=0)
